neutrons/mantid_total_scattering
# Autogrouping script

Implements a script to automatically generate a grouping file and mask by using KMeans or DBSCAN clustering algorithms as part of story [PD229](https://code.ornl.gov/sns-hfir-scse/diffraction/powder/powder-diffraction/-/issues/229).

**To test:**
The `autogrouping.py` script should be run with mantidpython and a json configuration file. An example configuration file and mask are attached
[autogrouping_config.json.txt](https://github.com/neutrons/mantid_total_scattering/files/6546763/autogrouping_config.json.txt)
[mask144974.out.txt](https://github.com/neutrons/mantid_total_scattering/files/6546774/mask144974.out.txt).

It is recommended to change the diamond_file to a cropped version for testing purposes since it can take a long time to generate results (especially using the "_CC" or "_DG" grouping methods)
